code/modules/reagents/reagent_dispenser.dm

```diff
@@ -117,17 +117,50 @@
 	icon_state = "fuel"
 	reagent_id = /datum/reagent/fuel
 	openable = TRUE
+	//an assembly attached to the tank
+	var/obj/item/assembly_holder/rig = null
```

Comment:
This will cause `rig` to hard delete if it is deleted while inside, from the looks of it.

You will want to override `/Exited` and check for if `exiting_atom == rig`, then set this to null.

Reply:
You also need to `QDEL_NULL` this in `Destroy()`.